File: auth-microservice/src/config/queueConfig.ts
import amqp from 'amqplib';
import fitnessLogger from '../libs/logger';

export const setupChannel = async (
  queueName: string,
  queueExchange: string,
  queueRk: string
) => {
  let retryCount = 4;

  let connection: amqp.Connection | null = null;
  let channel: amqp.Channel | null = null;
  try {
    connection = await amqp.connect('localhost');

    channel = await connection.createChannel();

    await channel.assertExchange(queueExchange, 'direct', { durable: true });

    await channel.assertQueue(queueName, { durable: true });

    await channel.bindQueue(queueName, queueExchange, queueRk);

    return {
      channel,
      connection,
    };
  } catch (err) {
    fitnessLogger.error(
      `Error creating an channel for ${queueName}, Retrying Again : ${retryCount}`
    );
    while (retryCount > 0) {
      fitnessLogger.info(`Retrying at count : ${retryCount}`);
      setupChannel(queueName, queueExchange, queueRk);
      retryCount = retryCount - 1;
    }

    if (retryCount.toString().startsWith('0')) {
      fitnessLogger.info(`Retrying count is finished`);
      process.exit(1);
    }
  }
};

File: model-microservice/config/queueConfig.py
import pika
import pika.channel
import pika.connection
import pika.exceptions
from pika.exchange_type import ExchangeType




def setup_channel(queue_name:str, queue_exchange:str,queue_rk:str):

    try:

        connection = pika.BlockingConnection(pika.ConnectionParameters('localhost'))
        channel = connection.channel()

        channel.exchange_declare(exchange=queue_exchange,exchange_type=ExchangeType.direct,durable=True)

        channel.queue_declare(queue=queue_name,durable=True)

        channel.queue_bind(queue=queue_name,exchange=queue_exchange,routing_key=queue_rk)

        return channel , connection
    
    except (pika.exceptions.AMQPError, Exception) as amqp_error:
        print(f'Error Occur in Amqp : {amqp_error}')